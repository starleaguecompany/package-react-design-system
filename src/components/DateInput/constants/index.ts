export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
