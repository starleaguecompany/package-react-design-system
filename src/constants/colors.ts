export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
