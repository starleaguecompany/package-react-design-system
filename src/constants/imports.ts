export const reactDSImportPath = '@starleaguecompany/react-design-system';
