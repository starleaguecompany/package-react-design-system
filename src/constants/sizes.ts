export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
