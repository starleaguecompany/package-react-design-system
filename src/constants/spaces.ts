export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
