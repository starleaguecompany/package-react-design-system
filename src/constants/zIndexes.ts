export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
