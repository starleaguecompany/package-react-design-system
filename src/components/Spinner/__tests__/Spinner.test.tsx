import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { CONTENT_SIZES } from '../../../constants/sizes'
import { Spinner } from '../../../index'

describe('Spinner', () => {
  test('should render correctly with no props', () => {
    render(<Spinner />)

    expect(screen.getByTestId('Spinner')).toMatchSnapshot()
    expect(screen.getByTestId('Spinner')).toHaveClass('container size-20', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<Spinner id="test-id" className="test-class" />)

    expect(screen.getByTestId('Spinner')).toMatchSnapshot()
    expect(screen.getByTestId('Spinner')).toHaveAttribute('id', 'test-id')
    expect(screen.getByTestId('Spinner')).toHaveClass('test-class')
  })

  test('should render correctly with size prop', () => {
    render(<Spinner size={CONTENT_SIZES.S24} />)

    expect(screen.getByTestId('Spinner')).toMatchSnapshot()
    expect(screen.getByTestId('Spinner')).toHaveClass('size-24')
  })

  test('should have svg inside ', () => {
    render(<Spinner size={CONTENT_SIZES.S24} />)

    expect(screen.getByTestId('Spinner')).toMatchSnapshot()
    expect(screen.getByTestId('Spinner').firstChild.nodeName).toEqual('svg')
  })
})
