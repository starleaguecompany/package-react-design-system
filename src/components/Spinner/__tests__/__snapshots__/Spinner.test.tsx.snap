// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should have svg inside  1`] = `
<span
  class="container size-24"
  data-qa="Spinner"
>
  <svg
    class="svg"
    viewBox="0 0 150 150"
  >
    <circle
      class="circle circle-1"
      cx="75"
      cy="75"
      r="60"
    />
    <circle
      class="circle circle-2"
      cx="75"
      cy="75"
      r="60"
    />
  </svg>
</span>
`;

exports[`Spinner should render correctly with attributes 1`] = `
<span
  class="test-class container size-20"
  data-qa="Spinner"
  id="test-id"
>
  <svg
    class="svg"
    viewBox="0 0 150 150"
  >
    <circle
      class="circle circle-1"
      cx="75"
      cy="75"
      r="60"
    />
    <circle
      class="circle circle-2"
      cx="75"
      cy="75"
      r="60"
    />
  </svg>
</span>
`;

exports[`Spinner should render correctly with no props 1`] = `
<span
  class="container size-20"
  data-qa="Spinner"
>
  <svg
    class="svg"
    viewBox="0 0 150 150"
  >
    <circle
      class="circle circle-1"
      cx="75"
      cy="75"
      r="60"
    />
    <circle
      class="circle circle-2"
      cx="75"
      cy="75"
      r="60"
    />
  </svg>
</span>
`;

exports[`Spinner should render correctly with size prop 1`] = `
<span
  class="container size-24"
  data-qa="Spinner"
>
  <svg
    class="svg"
    viewBox="0 0 150 150"
  >
    <circle
      class="circle circle-1"
      cx="75"
      cy="75"
      r="60"
    />
    <circle
      class="circle circle-2"
      cx="75"
      cy="75"
      r="60"
    />
  </svg>
</span>
`;
